cluster/get-kube-binaries.sh

```diff
@@ -0,0 +1,188 @@
+#!/bin/bash
+
+# Copyright 2016 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script downloads and installs the Kubernetes client and server binaries.
+# It is intended to be called from an extracted Kubernetes release tarball.
+#
+# We automatically choose the correct client binaries to download.
+#
+# Options:
+#  Set KUBERNETES_SERVER_ARCH to choose the server (Kubernetes cluster)
+#  architecture to download:
+#    * amd64 [default]
+#    * arm
+#    * arm64
+#    * ppc64le
+#
+#  Set KUBERNETES_SKIP_CONFIRM to skip the installation confirmation prompt.
+#  Set KUBERNETES_RELEASE_URL to choose where to download binaries from.
+#    (Default https://storage.googleapis.com/kubernetes-release/release).
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(cd $(dirname "${BASH_SOURCE}")/.. && pwd)
+
+KUBERNETES_RELEASE_URL="${KUBERNETES_RELEASE_URL:-https://storage.googleapis.com/kubernetes-release/release}"
+
+function detect_kube_release() {
+  if [[ ! -e "${KUBE_ROOT}/version" ]]; then
+    echo "Can't determine Kubernetes release." >&2
+    echo "This script should only be run from a prebuilt Kubernetes release." >&2
+    echo "Did you mean to use get-kube.sh instead?" >&2
+    exit 1
+  fi
+
+  KUBERNETES_RELEASE=$(cat "${KUBE_ROOT}/version")
+  DOWNLOAD_URL_PREFIX="${KUBERNETES_RELEASE_URL}/${KUBERNETES_RELEASE}"
+}
+
+function detect_client_info() {
+  local uname=$(uname)
+  if [[ "${uname}" == "Darwin" ]]; then
+    CLIENT_PLATFORM="darwin"
+  elif [[ "${uname}" == "Linux" ]]; then
+    CLIENT_PLATFORM="linux"
+  else
+    echo "Unknown, unsupported platform: (${uname})." >&2
+    echo "Supported platforms: Linux, Darwin." >&2
+    echo "Bailing out." >&2
+    exit 2
+  fi
+
+  local machine=$(uname -m)
+  if [[ "${machine}" == "x86_64" ]]; then
+    CLIENT_ARCH="amd64"
+  elif [[ "${machine}" == "i686" ]]; then
+    CLIENT_ARCH="386"
+  elif [[ "${machine}" == "arm" ]]; then
+    CLIENT_ARCH="arm"
+  elif [[ "${machine}" == "arm64" ]]; then
+    CLIENT_ARCH="arm64"
+  else
```

Comment:
The kubernetes release tarball doesn't include `hack`.

I wouldn't be surprised if this conditional is wrong. I copied it from `get-kube.sh`, which has been around for a while.